web/fluidity.money/src/components/RewardsInfoBox/RewardsInfoBox.tsx

```diff
@@ -46,6 +48,30 @@ const RewardsInfoBox = ({
     icon: <img src={imgLink(chain)} alt={`${chain}-icon`} />,
   }));
 
+  const [prizePool, setPrizePool] = useState<number>(Number(apiState.rewardPool.pools?.ethPool.toFixed(3)) || 0);
+
+  useEffect(() => {
+    chain === `ETH` && 
+    setPrizePool(Number(apiState.rewardPool.pools?.ethPool.toFixed(3)) || 0);
+
+    chain === `SOL` && 
+    setPrizePool(Number(apiState.rewardPool.pools?.solPool.toFixed(3)) || 0);
+
+  },[apiState.rewardPool.pools?.ethPool, apiState.rewardPool.pools?.solPool, chain]);
+  
+  useEffect(() => {
+    if (!apiState.rewardPool.loading) return
+
+    const interval = setInterval(() => {
+      setPrizePool((prizePool) => {
+        const random = Math.random() * 999999
+        return Number(random.toFixed(3))
+      })
+    } , 200)
+
+    return () => clearInterval(interval)
+  }, [apiState.rewardPool.loading])
+
```

Comment:
Same with this, can be done on parent component (up one level).  if the idea was because of the loading state ?, you can still pass that in here as just data